scripts/checkout-part.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/python3
```

Comment:
A small part of me feels bad that we're reimplementing submodules, but then I remember that I don't particularly like submodules.

I will console myself by merely modifying snapcraft.yaml instead of remembering submodules commands.

Reply:
I've certainly never entirely 'got' how submodules are expected to work. And given that we need to specifiy the commits in snapcraft.yaml for snapcraft, maybe this is OK...

Reply:
Although hmm it does seem that snapcraft natively supports submodules. So we could move to that instead if we wanted to?

Reply:
I will be fine with submodules when we choose to move to them, really I'm just disappointed with how non-transparent the git submodule implementation is.